lib/src/query_component.dart

```diff
@@ -83,6 +107,25 @@ class QueryComponent implements OnInit {
   /// Used in [query_component.html] as part of the radio button id.
   String getQueryTypeShortName(QueryType queryType) => queryType.shortName;
 
+  bool _missingRequiredParameters() {
+    switch (queryType) {
+      case QueryType.byAdvertiser:
+        return advertiserId == null || advertiserId.isEmpty;
+
+      case QueryType.byMediaPlan:
```

Comment:
can i click even when i have alphabet text entered? the alert will show but is there anything stopping from the request being made?

Reply:
Ohh good catch, maybe I should disable the button until all inputs are filled in & valid. Seems like the error message should be used when I get to error handling for failed query request.